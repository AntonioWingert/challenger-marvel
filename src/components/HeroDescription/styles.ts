import { styled } from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  min-height: 100vh;
  height: auto;
  background-color: #ccc;
  border-radius: 0.5rem;

  h2 {
    font-size: 4rem;
    text-align: center;
  }

  img {
    width: auto;
    height: auto;
  }

  p {
    font-size: 1rem;
    text-align: center;
    width: 40%;
  }

  a {
    display: flex;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    color: ${({theme}) => theme.colors["secondary"]};
    width: 35rem;
    height: 4rem;
    background-color: ${({theme}) => theme.colors["bg-primary"]};
  }

  @media (max-width: 768px) {

    a {
      width: 100%;
      height: 2rem;
    }

    img {
      width: 80%;
      height: 40vh;
    }

    p {
      width: 80%;
      text-align: justify;
    }
  }
`;