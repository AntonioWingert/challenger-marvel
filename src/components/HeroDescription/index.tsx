import { Link } from 'react-router-dom';
import * as S from './styles';

type HeroProps = {
  id: number;
  name: string;
  thumbnail: string;
}

export default function HeroDescription({ id = 0, name = '', thumbnail = ''}: HeroProps) {
  return (
    <S.Container>
      <h2>{name}</h2>
      <img src={`${thumbnail}/portrait_medium.jpg`} alt={`Hero image ${name}`} />
      <Link to={`/description/${id}`}>Ver Mais</Link>
    </S.Container>
  );
}