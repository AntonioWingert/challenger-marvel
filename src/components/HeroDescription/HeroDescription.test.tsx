import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import HeroDescription from '.';

describe('<HeroDescription />', () => {
  it('should render correctly with default props', () => {
    const { container } = render(
      <HeroDescription description='Description' name='Hero' thumbnail='Hero.Image' />
    );

    expect(screen.getByText(/hero/i)).toBeVisible();
    expect(screen.getByAltText('Hero image Hero')).toBeVisible();
    expect(screen.getByAltText('Hero image Hero')).toHaveAttribute('src', 'Hero.Image/portrait_uncanny.jpg');
    expect(container.firstChild).toMatchSnapshot();
  });
});