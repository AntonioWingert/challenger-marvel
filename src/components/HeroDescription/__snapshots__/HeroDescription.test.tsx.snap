// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeroDescription /> > should render correctly with default props 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  min-height: 100vh;
  height: auto;
  background-color: #ccc;
  border-radius: 0.5rem;
}

.c0 h2 {
  font-size: 4rem;
  text-align: center;
}

.c0 img {
  width: auto;
  height: auto;
}

.c0 p {
  font-size: 1rem;
  text-align: center;
  width: 40%;
}

.c0 a {
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  color: #F5F5F5;
  width: 35rem;
  height: 4rem;
  background-color: #4A1414;
}

@media (max-width: 768px) {
  .c0 a {
    width: 100%;
    height: 2rem;
  }

  .c0 img {
    width: 80%;
    height: 40vh;
  }

  .c0 p {
    width: 80%;
    text-align: justify;
  }
}

<div
  class="c0"
>
  <h2>
    Hero
  </h2>
  <img
    alt="Hero image Hero"
    src="Hero.Image/portrait_uncanny.jpg"
  />
  <p>
    Description
  </p>
  <a
    href="/"
  >
    Voltar
  </a>
</div>
`;
